nemo_curator/utils/file_utils.py

```diff
@@ -45,7 +46,32 @@ def expand_outdir_and_mkdir(outdir):
     return outdir
 
 
-def get_all_files_paths_under(root, recurse_subdirectories=True, followlinks=False):
+def filter_files_by_extension(
+    files_list: str,
+    filter_by: Union[str, List[str]],
+):
+    filtered_files = []
+
+    if isinstance(filter_by, str):
+        filter_by = [filter_by]
+
+    file_extensions = [s if s.startswith(".") else "." + s for s in filter_by]
+
+    for file in files_list:
+        if file.endswith(tuple(file_extensions)):
+            filtered_files.append(file)
+        else:
+            warnings.warn(f"Skipping read for file: {file}")
```

Comment:
I wonder if this might get too noisy in some cases. I'm leaning towards warning once if we have to skip, but not for every file we skip.